packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -487,16 +518,15 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 
                     // It's possible we transitioned to offline flow while waiting for this op.
                     if (pendingBlobEntry.status === PendingBlobStatus.OnlinePendingOp) {
-                        pendingBlobEntry.handleP.resolve(this.getBlobHandle(message.metadata.blobId));
```

Comment:
I think this check should be removed, since the point was to wait for the second op w/ localID if we transitioned to offline flow after sending an op without it. Since we always send an op with local ID now, `transitionToOffline()` should only send an op if one hasn't been sent already, and this should always resolve regardless of status.

Reply:
Okay. So in transitionToOffline(), only send an op if opsInFlight does not have an entry for this local Id <-> storageId combination.
This check does not have to be removed because `this.deleteAndEmitsIfEmpty(message.metadata.localId);` outside the if (waitingBlobs !== undefined) will take care of it. We need that check for ops that were sent in disconnected mode anyway.